tests/storage/restore.go

```diff
@@ -395,6 +468,100 @@ var _ = SIGDescribe("[Serial]VirtualMachineRestore Tests", func() {
 				deleteRestore(r2)
 				deleteRestore(restore)
 			})
+
+			It("should successfully restore to a different VM", func() {
```

Comment:
Looks like we are restoring a snapshot of VM A to and existing VM B.  If I recall, we wanted this to fail.

Reply:
IMO it could be very useful to take a snapshot once and then restore to any VM that fits. For example, I can see this being very useful with VMPools.

Would shouldn't we allow such a thing?

Reply:
Added a test that verifies failure in this case